server/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -681,7 +681,7 @@ public String toString() {
             }
             builder.append(' ');
         }
-        return builder.append(ExceptionsHelper.detailedMessage(this).trim()).toString();
+        return builder.append(super.toString().trim()).toString();
```

Comment:
Won't we lose all stack information by just using the exception tostring?

Reply:
Only if whatever's calling `ElasticsearchException.toString()` isn't also printing the stack trace. The problem with calling `ElasticsearchException.stackTrace()` here is that it calls `.toString()` on the exception, so you end up blowing the stack. `detailedMessage()` didn't do this because it essentially reimplemented `Throwable.toString()`. Open to ideas here.